### ISSUE TITLE
Cannot find module '@amcharts/amcharts4'

### ISSUE THREAD

**Issue by eggsby** [amcharts/amcharts4#1818]
When I try to run through my build system, (gulp/babelify/browserify) I get:

```[16:45:53] Error: Cannot find module '@amcharts/amcharts4' from .....```

My file includes:

```
import * as am4core from "@amcharts/amcharts4/core";
```

The install is definitely there in the local node_modules directory, but it looks like there is some funny business going on in the way amcharts4 is defining its exports. Importing worked fine in previously published amcharts libraries, for example "@amcharts/amcharts3-react". 

Digging a little into the install, I notice core.js begins:

```
export { System, system } from "./.internal/core/System"
```

But this looks like an experimental module format not supported by node or npm.

If you point the core.js at the latest version of node you get:

```
export { System, system } from "./.internal/core/System";
^^^^^^

SyntaxError: Unexpected token 'export'
```

**Comment by Pauan** [comment]
There is nothing wrong with the amCharts code, it is using [100% standard ES6 modules](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Modules) (which are supported by Webpack, Babel, Rollup, Parcel, Angular, React, and *all* modern browsers)

There is something wrong with your build setup, could you please post your Gulp file?

**Comment by eggsby** [comment]
Here is a minimal gulp config that can reproduce the error:
@ latest version of all required deps

```
let gulp = require('gulp');
let browserify = require('browserify');
let source = require('vinyl-source-stream');
let uglify = require('gulp-uglify');
let streamify = require('gulp-streamify');
let tsify = require('tsify');
let _ = require("underscore");

let packageJson = require('./package.json');

let dependencies =
    // these dependencies include @amcharts/amcharts4
    Object.keys(packageJson && packageJson.dependencies || {});

let baseBrowserifyConfig = {
    extensions: ['.ts', '.tsx', '.js', '.jsx'],
    debug: true,
    cache: {},
    packageCache: {},
    paths: ["src/js"],
    transform: [["babelify", {
        "presets": ["@babel/preset-react", "@babel/preset-env"],
        "plugins": ["@babel/plugin-proposal-class-properties", "@babel/plugin-syntax-dynamic-import"],
        "extensions": [".ts", ".tsx", ".js", ".jsx"],
        "global": true
    }]],
    plugin: [tsify]
};

function bundleLibs() {
    return browserify(baseBrowserifyConfig)
        .require(dependencies)
        .bundle()
        .pipe(source("libs.js"))
        .pipe(streamify(uglify()))
        .pipe(gulp.dest("webroot/dist"));
}

function bundle() {
    let bundleBrowserifyConfig =
        _.extend(baseBrowserifyConfig, {entries: 'src/js/entry.js'});
    return browserify(bundleBrowserifyConfig).external(dependencies)
        .bundle()
        .pipe(source("bundle.js"))
        .pipe(streamify(uglify()))
        .pipe(gulp.dest("webroot/dist"));
}

gulp.task('default', gulp.parallel(bundleLibs, bundle));
```

Specifically when we say `browserify.require(amchartsDependency)`

It looks like this may be related to a lack of browserify support for es6 modules, since they are currently experimental in node (--experimental-modules) - even then node expects them to have the .mjs extension. 

Looks to be related to https://github.com/browserify/browserify/issues/1186#issuecomment-438793510

To summarize, it looks like amcharts4 does not work with browserify bundling because browserify expects standard modules.

**Comment by eggsby** [comment]
I'm going to close this issue because amcharts is just publishing valid es6 modules. 
It is browserify that does not support es6 modules.

**Comment by Pauan** [comment]
@eggsby Babel supports converting ES6 modules into CommonJS modules (which Browserify supports).

You should try changing your config to this:

```js
function bundleLibs() {
    return browserify({
        extensions: ['.ts', '.tsx', '.js', '.jsx'],
        debug: true,
    })
        .transform("babelify", {
            "presets": [
                ["@babel/preset-react", { "modules": "cjs" }],
                "@babel/preset-env"
            ],
            "plugins": [
                "@babel/plugin-proposal-class-properties",
                "@babel/plugin-syntax-dynamic-import"
            ],
            "extensions": [".ts", ".tsx", ".js", ".jsx"],
            "global": true
        })
        .plugin(tsify)
        .bundle()
        .pipe(source("libs.js"))
        .pipe(streamify(uglify()))
        .pipe(gulp.dest("webroot/dist"));
}
```

(It shouldn't be necessary to use `.require(dependencies)`, Browserify is able to find the dependencies itself)
